content/utilities/favourite-areas/plugin.js

```diff
@@ -0,0 +1,165 @@
+// Favourite Areas
+// Save client width, height and coordinates of currently visible area
+
+import {
+  html,
+  render,
+  useState,
+  useEffect,
+} from "https://unpkg.com/htm/preact/standalone.module.js";
+
+// Used to give a rough estimate of zoom level
+const maxClientSize = 1000000;
+
+const input = {
+  width: "250px",
+  padding: "5px",
+  outline: "none",
+  color: "black",
+};
+
+const row = {
+  display: "flex",
+  padding: "3px",
+};
+
+const goToButton = {
+  "margin-left": "auto",
+  "margin-right": "5px",
+};
+
+const addButton = {
+  "margin-left": "auto",
+};
+
+const zoom = {
+  "margin-left": "5px",
+};
+
+function AreaRow({ area, onRemove, onGoTo }) {
+  const removeClicked = () => {
+    onRemove(area.name);
+  };
+
+  const goToClicked = () => {
+    onGoTo(area.name);
+  };
+
+  const calculateZoom = (area) => {
+    const zoom = maxClientSize / ((area.width + area.height) / 2);
+    return Math.round(zoom * 10) / 10;
+  };
+
+  const trimCoordinate = (coordinate) => {
+    return coordinate.toString().split(".")[0];
+  };
+
+  return html`<div style=${row}>
+    <span
+      >${area.name} (${trimCoordinate(area.x)}, ${trimCoordinate(area.y)})
+    </span>
+    <span style=${zoom}>${calculateZoom(area)}x</span>
+    <button style=${goToButton} onClick=${goToClicked}>Go To</button>
+    <button onClick=${removeClicked}>X</button>
+  </div> `;
+}
+
+function App() {
+  const [areaName, setAreaName] = useState(null);
+  const [areas, setAreas] = useState([]);
+
+  useEffect(() => {
+    const areasJson = localStorage.getItem("favourite-areas");
+    const cachedAreas = JSON.parse(areasJson);
```

Comment:
On first load this plugin failed for me, and actually requried a game refresh to get it unstuck. I think you need to check to see if areasJson is defined and provide an empty default maybe

Reply:
Ooops! That sounds about right, I'll fix it

Reply:
@Robiquet sorry to be the bearer of bad news, but we don't allow plugins (with 1 rare exception being repeat-attack) to access localstorage. This is where a user's private key is stored and we don't want plugins mucking around in there. Can you remove this feature or make it an import/export-to-json type of thing?